src/kubernetes.rs

```diff
@@ -22,8 +23,23 @@ pub struct Kubernetes {
 impl Kubernetes {
     pub fn connect(kubeconfig_path: Option<String>, scheme: Option<String>, host: Option<String>, port: Option<u32>, search_uri: bool) -> Result<Kubernetes, KubernetesError> {
         let kubeconfig = KubeConfig::load(kubeconfig_path);
+        let token_file = "/var/run/secrets/kubernetes.io/serviceaccount/token";
         let http_client;
-        if let Ok(conf) = &kubeconfig {
+
+        if std::path::Path::new(token_file).exists() {
```

Comment:
We first check if there is a service account token. If not we try the kubeconfig.

This means in the daemonset we don't need to mount the kubeconfig as a file.